main.cpp

```diff
@@ -12,13 +12,26 @@ int main(int argc, char* argv[])
     QCoreApplication app(argc, argv);
     auto args = app.arguments();
     bool use_gpu = true;
+    int gpu_bitcount = 0;
+
+    int first_file_idx = 1;
+    while(args[first_file_idx].startsWith("--")) first_file_idx++;
```

Comment:
This is no longer a good idea. Since we can use "--gpu_bitcount 0" as an option, the count would stop at the value, and we would not get the proper filename.

Also, the value of "first_file_idx" is updated again inside the check for "--gpu_bitcount".